### PR TITLE
Updated APIRequestor.cs with extra checks

### PR DESCRIPTION
**Pull request by Gusted**
Added extra checks of URL is valid or not

### CODE REVIEW

**Comment by Gusted** [comment]
@KieranBond I think this are all error's that cause error

File: MiniSpotify/MiniSpotify/Source/APIRequestor.cs [hunk 1]
```diff
@@ -52,6 +52,9 @@ public static APIRequestor Instance
         private string m_clientID = "93f2598a9eaf4056b34f7b5ca254ff17";
         private string m_clientSecret = "";
         private string m_clientSecretPath = "\\Assets\\Files\\ClientSecret.txt";
+        
+        private string t_albumimage = "";
```
Comment (KieranBond) [inline MiniSpotify/MiniSpotify/Source/APIRequestor.cs]:
```suggestion
        private string m_albumimage = "";
```

Let's make this fit with the coding standards followed in the rest of the file please.

File: MiniSpotify/MiniSpotify/Source/APIRequestor.cs [hunk 2]
```diff
@@ -52,6 +52,9 @@ public static APIRequestor Instance
...
+        public static string t_id = "";
```
Comment (KieranBond) [inline MiniSpotify/MiniSpotify/Source/APIRequestor.cs]:
```suggestion
        public static string m_id = "";
```
Same with this one, let's keep the standards up.

Also, what's the reasoning behind using a static variable here?

File: MiniSpotify/MiniSpotify/Source/APIRequestor.cs [hunk 3]
```diff
@@ -209,15 +212,31 @@ public void ModifyPlayback()
...
+                {
+                    try
+                    {
+                        m_spotifyWebAPI.SkipPlaybackToPrevious();
+                    }
+                    catch (ArgumentException e) // Spammed probaly the button
+                    {
+                        Console.WriteLine(e.StackTrace);
+                    }
+                }
```
Comment (KieranBond) [inline MiniSpotify/MiniSpotify/Source/APIRequestor.cs:238]: I'm happy with this for now, it's better than what I was doing.

I'll modify this later, I think I want to add some kind of log functionality so we can have crash logs.

File: MiniSpotify/MiniSpotify/Source/APIRequestor.cs [hunk 4]
```diff
@@ -242,13 +261,17 @@ public string GetCurrentSongArtwork()
...
-                    imageURL = m_spotifyWebAPI.GetPlayingTrack().Item.Album.Images[0].Url;
+                    try
+                    {
+                        imageURL = m_spotifyWebAPI.GetPlayingTrack().Item.Album.Images[0].Url;
+                        t_albumimage = imageURL; //Set it as 'backup'
+                    }catch(Exception e)
+                    {
+                        imageURL = t_albumimage; //Internet failure and get last time album as backup
+                        Console.WriteLine(e.StackTrace);
+                    }
```
Comment (KieranBond) [inline MiniSpotify/MiniSpotify/Source/APIRequestor.cs:273]: Two points here;
* I'm not a fan of us assuming it's an internet failure in the comment, but that's fine for now - It's not important why it fails most of the time.

* We don't do anything about if there is an error, we're now just ignoring it completely. Let's make this atleast do a Console output for now.

File: MiniSpotify/MiniSpotify/Source/APIRequestor.cs [hunk 5]
```diff
@@ -284,9 +307,15 @@ public float GetLatestSongProgress()
...
-                return progress;
+                try
+                {
+                    float progress = (float)m_spotifyWebAPI.GetPlayback().ProgressMs / (float)currentTrack.DurationMs;
+                    return progress;
+                }catch(NullReferenceException e) //Unexpected spotify closed
+                {
+                    Console.WriteLine(e.StackTrace);
+                    return -1;
+                }
```
Comment (KieranBond) [inline MiniSpotify/MiniSpotify/Source/APIRequestor.cs:317]: I'm happy with this.

**Comment by Gusted** [comment]
Updated t_ to m_ and edited the comments to a more 'stable' comment

File: MiniSpotify/MiniSpotify/Source/MainWindow.xaml.cs [hunk 6]
```diff
@@ -51,8 +51,18 @@ public void UpdateUI(FullTrack a_latestTrack = null)
...
+                {
+                    if (APIRequestor.Instance.GetLatestTrack().Id != null)
+                    {
+                        artworkURL = APIRequestor.Instance.GetLatestTrack().Id;//Get the track ID
+                        APIRequestor.t_id = artworkURL; //Setting the temp id to current one
+                    }
+                }catch(NullReferenceException e)// Internet connection Failure of not getting
+                {
+                    Console.WriteLine(e.StackTrace);
+                    artworkURL = APIRequestor.t_id; //Getting backup of last used id
```
Comment (KieranBond) [inline MiniSpotify/MiniSpotify/Source/MainWindow.xaml.cs]: I don't like this. Whilst yes we're handling there being an error, we have no guarantee that value is safe, nor do I like us assuming it's not garbage. I'd rather we didn't set anything, and just logged an error here.

Comment (Gusted) [reply MiniSpotify/MiniSpotify/Source/MainWindow.xaml.cs]: Okay let me change it to return it null.

Comment (Gusted) [reply MiniSpotify/MiniSpotify/Source/MainWindow.xaml.cs]: I fixed it with the last commit

File: MiniSpotify/MiniSpotify/Source/APIRequestor.cs (hunk 2 above)
Comment (KieranBond) [reply MiniSpotify/MiniSpotify/Source/APIRequestor.cs]: We don't use this variable anymore, please remove this.

File: MiniSpotify/MiniSpotify/Source/APIRequestor.cs (hunk 4 above)
Comment (Gusted) [reply MiniSpotify/MiniSpotify/Source/APIRequestor.cs:273]: Comment is replaced and better stable explanation and removed the 'backup'

File: MiniSpotify/MiniSpotify/Source/APIRequestor.cs (hunk 2 above)
Comment (Gusted) [reply MiniSpotify/MiniSpotify/Source/APIRequestor.cs]: Fixed it with the last commit.

**Comment by Gusted** [comment]
Updated all comments and removed the unused variable

**Review by KieranBond — APPROVED** [review]
This is pretty good. It doesn't go and completely solve our issue of handling errors, but it makes the right step forward towards handling them gracefully and should hopefully fix some bugs.

**Pull request merged**